### PR TITLE
docs: add concurent compaction docs

### PR DESCRIPTION
**Pull request by 317brian**
### Description
* Adds the docs for concurrent compaction
* Refactors the page to be more structured:

General compaction info as the parent w/ manual compaction and auto compaction as its children:
image

Build preview: https://druid-git-concurrent-compaction-317brian.vercel.app/docs/latest/data-management/compaction

<hr>

This PR has:

- [x] been self-reviewed.

### CODE REVIEW

File: docs/data-management/automatic-compaction.md [hunk 1]
```diff
@@ -203,6 +203,82 @@ The following auto-compaction configuration compacts updates the `wikipedia` seg
 }
 ```
 
+## Concurrent compaction
+
+:::info
+Concurrent compaction is an [experimental feature](../development/experimental.md) and is not currently available for SQL-based ingestion.
```
Comment (317brian) [inline docs/data-management/automatic-compaction.md:209]: @AmatyaAvadhanula will concurrent compaction w/ MSQ be in for 28 or no?

File: docs/data-management/automatic-compaction.md [hunk 2]
```diff
@@ -203,6 +203,82 @@ The following auto-compaction configuration compacts updates the `wikipedia` seg
...
+
+```json
+"useSharedLock": "true"
+```
+
+Keep in mind that `taskLockType` takes precedence over `useSharedLock`. Do not use it with `REPLACE` task locks.
+
+##### Add a task lock using the Druid console
+
+As part of the batch or streaming ingestion **Load data** wizards, you can, you can choose **Allow concurrent append tasks** to use concurrent compaction.
```
Comment (317brian) [inline docs/data-management/automatic-compaction.md:279]:
```suggestion
As part of the batch or streaming ingestion **Load data** wizards, you can choose **Allow concurrent append tasks** to use concurrent compaction. 
```

File: docs/data-management/automatic-compaction.md (hunk 1 above)
Comment (317brian) [reply docs/data-management/automatic-compaction.md:209]: Answered elsewhere. It won't be.

File: docs/data-management/automatic-compaction.md [hunk 3]
```diff
@@ -203,6 +203,82 @@ The following auto-compaction configuration compacts updates the `wikipedia` seg
...
+    "dataSource": "YOUR_DATASOURCE",
+    "taskContext": {
+        "taskLockType": "REPLACE"
+    }
+}'
+```
+
+##### Update the compaction settings with the UI
+
+In the **Compaction config** for a datasource, set  **Allow concurrent compaction append tasks** to **True**.
```
Comment (AmatyaAvadhanula) [inline docs/data-management/automatic-compaction.md:243]: The UI for compaction says "Allow concurrent compactions (experimental)"
image

Comment (317brian) [inline docs/data-management/automatic-compaction.md:243]:
```suggestion
In the **Compaction config** for a datasource, set  **Allow concurrent compactions (experimental)** to **True**.
```

File: docs/data-management/automatic-compaction.md [hunk 4]
```diff
@@ -203,6 +203,85 @@ The following auto-compaction configuration compacts updates the `wikipedia` seg
...
+
+##### Update the compaction settings with the UI
+
+In the **Compaction config** for a datasource, set  **Allow concurrent compactions (experimental)** to **True**.
+
+#### Add a task lock type to your ingestion job
+
+Next, you need to configure the task lock type for your ingestion job. For streaming jobs, the context parameter goes in your supervisor spec. For legacy JSON-based batch ingestion, the context parameter goes in your ingestion spec. You can provide the context parameter through the API like any other parameter for a streaming ingestion or JSON-based batch ingestion or UI.
+
+##### Add the task lock type through the API
```
Comment (AmatyaAvadhanula) [inline docs/data-management/automatic-compaction.md:249]: Could we please explicitly add that a supervisor spec must always have an APPEND lock when using concurrent append and replace?

File: docs/data-management/automatic-compaction.md [hunk 5]
```diff
@@ -203,6 +203,85 @@ The following auto-compaction configuration compacts updates the `wikipedia` seg
 }
 ```
 
+## Concurrent append and replace
+
+:::info
+Concurrent append and replace is an [experimental feature](../development/experimental.md) and is not currently available for SQL-based ingestion.
+:::
+
+If you enable automatic compaction, you can use concurrent append and replace to concurrently compact data as you ingest it for streaming and legacy JSON-based batch ingestion.
```
Comment (kfaraz) [inline docs/data-management/automatic-compaction.md:212]:
```suggestion
This feature allows you to safely replace the existing data in an interval of a datasource while new data is being appended to that interval. One of the most common applications of this is appending new data (using say streaming ingestion) to an interval while compaction of that interval is already in progress.
```

File: docs/data-management/automatic-compaction.md [hunk 6]
```diff
@@ -203,6 +203,85 @@ The following auto-compaction configuration compacts updates the `wikipedia` seg
...
+:::
+
+If you enable automatic compaction, you can use concurrent append and replace to concurrently compact data as you ingest it for streaming and legacy JSON-based batch ingestion. 
+
+Setting up concurrent append and replace is a two-step process. The first is to update your datasource and the second is to update your ingestion job.
+
+Using concurrent append and replace in the following scenarios can be beneficial:
+
+- If the job with an `APPEND` task and the job with a `REPLACE` task have the same segment granularity. For example, when a datasource and its streaming ingestion job have the same granularity.
+- If the job with an `APPEND` task  has a finer segment granularity than the replacing job.
```
Comment (kfaraz) [inline docs/data-management/automatic-compaction.md:219]:
```suggestion
You can enable concurrent append and replace by ensuring the following:
- The append task (with `appendToExisting` set to `true`) has `taskLockType` set to `APPEND` in the task context.
- The replace task (with `appendToExisting` set to `false`) has `taskLockType` set to `REPLACE` in the task context.
- The segment granularity of the append task is equal to or finer than the segment granularity of the replace task.
```

File: docs/data-management/automatic-compaction.md [hunk 7]
```diff
@@ -203,6 +203,85 @@ The following auto-compaction configuration compacts updates the `wikipedia` seg
...
+
+We do not recommend using concurrent append and replace when the job with an `APPEND` task has a coarser granularity than the job with a `REPLACE` task. For example, if the `APPEND` job has a yearly granularity and the `REPLACE` job has a monthly granularity. The job that finishes second will fail.
```
Comment (kfaraz) [inline docs/data-management/automatic-compaction.md:221]: This point should be in a note or warning block.

Two more points to call out are that:

```
At any point in time
- There can only be a single task that holds a `REPLACE` lock on a given interval of a datasource.
- There may be multiple tasks that hold `APPEND` locks on a given interval of a datasource and append data to that interval simultaneously.
```

**Review by kfaraz — COMMENTED** [review]
Thanks for the changes, @317brian ! I have left some suggestions.

Given that this is an experimental feature, we can proceed with the docs that we have so far.

As the feature evolves and is hardened, we will enrich these docs to include some more technical details.

File: docs/data-management/automatic-compaction.md [hunk 8]
```diff
@@ -203,6 +203,106 @@ The following auto-compaction configuration compacts updates the `wikipedia` seg
...
+This feature allows you to safely replace the existing data in an interval of a datasource while new data is being appended to that interval. One of the most common applications of this is appending new data (using say streaming ingestion) to an interval while compaction of that interval is already in progress.
+
+To set up concurrent append and replace, you need to ensure that your ingestion jobs have the appropriate lock types:
+
+You can enable concurrent append and replace by ensuring the following:
+- The append task (with `appendToExisting` set to `true`) has `taskLockType` set to `APPEND` in the task context.
+- The replace task (with `appendToExisting` set to `false`) has `taskLockType` set to `REPLACE` in the task context.
+- The segment granularity of the append task is equal to or finer than the segment granularity of the replace task.
+
+:::info
```
Comment (317brian) [inline docs/data-management/automatic-compaction.md:221]: renders as 

image

File: docs/data-management/automatic-compaction.md (hunk 4 above)
Comment (317brian) [reply docs/data-management/automatic-compaction.md:265]: Added below in the append section that talks about lock type

**Comment by 317brian** [comment]
@kfaraz and @AmatyaAvadhanula I think I got all your comments. I manually made the changes  for the comment suggestions since I already had the page open

File: docs/data-management/automatic-compaction.md [hunk 9]
```diff
@@ -203,6 +203,106 @@ The following auto-compaction configuration compacts updates the `wikipedia` seg
...
+
+When using concurrent append and replace, keep the following in mind:
+
+- When the job with an `APPEND` task has a coarser granularity than the job with a `REPLACE` task, you shouldn't use concurrent append and replace. For example, if the `APPEND` job has a yearly granularity and the `REPLACE` job has a monthly granularity. The job that finishes second will fail.
```
Comment (kfaraz) [inline docs/data-management/automatic-compaction.md:225]:
```suggestion
- Concurrent append and replace fails if the task with `APPEND` lock uses a coarser segment granularity than the task with the `REPLACE` lock. For example, if the `APPEND` task uses a segment granularity of YEAR and the `REPLACE` task uses a segment granularity of MONTH, you should not use concurrent append and replace.
```

**Review by kfaraz — APPROVED** [review]
Minor change, otherwise LGTM.

**Review by AmatyaAvadhanula — APPROVED** [review]

**Pull request merged**